suzieq/engines/pandas/interfaces.py

```diff
@@ -100,55 +100,85 @@ def _check_vlan_match(self, vlan_list: List[str],
 
-        for vlan in vlan_list:
-            if vlan.startswith(('<=', '>=')):
-                op = vlan[0:2]
-                vlan = vlan[2:].strip()
-            elif vlan.startswith(('<', '>', '!')):
-                op = vlan[0]
-                vlan = vlan[1:].strip()
-            else:
-                op = '=='
-
-            vlan = int(vlan)
-            if op == "!":
-                tmpdf = df[df.apply(
-                    lambda row, vlan: all(opdict[op](v, vlan)
-                                          for v in row.vlanList) and
-                    opdict[op](row.vlan, vlan), args=(vlan,), axis=1)]
-            else:
-                tmpdf = df[df.apply(
-                    lambda x, vlan: any(opdict[op](y, vlan)
-                                        for y in x.vlanList) or
-                    opdict[op](x.vlan, vlan), args=(vlan,), axis=1)]
-            if cond == "or":
-                resdf_list.append(tmpdf.reset_index(drop=True))
-            else:
-                df = tmpdf
-
-        if resdf_list:
-            df = pd.concat(resdf_list)
+        def _check_cond(vlist: pd.Series, op: Callable, *args):
+            fn = opdict[op]
+            return vlist.apply(lambda ls: any(fn(vlan, *args) for vlan in ls))
+
+        def _interval(ival, start_val, start_op, end_val, end_op):
+            start_op = opdict[start_op]
+            end_op = opdict[end_op]
+            return start_op(ival, start_val) and end_op(ival, end_val)
+
+        def extract_op(expression: Union[str, int]) -> Tuple[str, int]:
+            op = '=='
+            val = expression
+            if isinstance(expression, str):
+                if expression.startswith(('<=', '>=')):
+                    val = expression[2:]
+                    op = expression[:2]
+                elif expression.startswith(('<', '>')):
+                    val = expression[1:]
+                    op = expression[:1]
+                elif expression.startswith('!'):
+                    val = expression[1:]
+                    op = expression[:1]
+            return op, val
+
+        if not vlan_filters:
+            return df
 
-        return df.query('vlanList.str.len() > 0 or vlan != 0')
+        opdict.update({'><': _interval})
+        not_filters = []
+        match_filters = []
+
+        i = 0
+        while i < len(vlan_filters):
+            op, fval = extract_op(vlan_filters[i])
+            if op.startswith('!'):
+                not_filters.append(
+                    f'(vlan != {fval} '
+                    f'and ~@_check_cond(vlanList, "==", {fval}))')
+            elif (op.startswith('>')
+                  and i+1 < len(vlan_filters)
+                  and isinstance(vlan_filters[i+1], str)
+                  and vlan_filters[i+1].startswith('<')):
+                # In this case we are checking if the user asked for an
+                # an interval. So if we find a sequence of > and <,
+                # we will combine the rules
```

Comment:
Just to be sure I understood correctly:
- `>10 <50` is equal to `10 < x < 50`
- `<50 >10` is equal to `x < 50 or x > 10` (which is everything)

Correct?

Reply:
Yes, at the moment we detect an interval only when the rule is written following the right order